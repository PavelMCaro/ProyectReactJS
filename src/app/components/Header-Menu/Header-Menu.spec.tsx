import * as React from 'react';
import * as Enzyme from 'enzyme'
import { Menu } from './index';
import { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Test Header-Menu Component', () => {

    const props = {
        imgSrc: "ruta/imagen",
        userLogin: "Pavel",
        showModal: true,
        showAvatar: true,
        showButton: true
    }
    const component = shallow((
        <Menu {...props} /> 
    ));
    it('Test Show Component', () =>{
        console.log(component.debug());
        expect(component).toMatchSnapshot();
    });
});