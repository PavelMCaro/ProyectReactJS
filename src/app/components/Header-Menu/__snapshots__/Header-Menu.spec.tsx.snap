// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header-Menu Component Test Show Component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    imgSrc="ruta/imagen"
    showAvatar={true}
    showButton={true}
    showModal={true}
    userLogin="Pavel"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.img
          showAvatar={true}
          src="ruta/imagen"
        />,
        <styled.p>
          Pavel
        </styled.p>,
        <styled.button
          onClick={true}
          showButton={true}
        >
          Ingresar
        </styled.button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "showAvatar": true,
          "src": "ruta/imagen",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Pavel",
        },
        "ref": null,
        "rendered": "Pavel",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Ingresar",
          "onClick": true,
          "showButton": true,
        },
        "ref": null,
        "rendered": "Ingresar",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.img
            showAvatar={true}
            src="ruta/imagen"
          />,
          <styled.p>
            Pavel
          </styled.p>,
          <styled.button
            onClick={true}
            showButton={true}
          >
            Ingresar
          </styled.button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "showAvatar": true,
            "src": "ruta/imagen",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Pavel",
          },
          "ref": null,
          "rendered": "Pavel",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Ingresar",
            "onClick": true,
            "showButton": true,
          },
          "ref": null,
          "rendered": "Ingresar",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
